GlucoseOnFhir/ObservationViewController.swift

```diff
@@ -0,0 +1,170 @@
+                cell.detailTextLabel!.text = "value"
+            case 1:
+                cell.textLabel!.text = self.observation.valueQuantity?.unit?.description
+                cell.detailTextLabel!.text = "unit"
+            case 2:
+                cell.textLabel!.text = self.observation.valueQuantity?.system?.description
+                cell.detailTextLabel!.text = "system"
+            case 3:
+                cell.textLabel!.text = self.observation.valueQuantity?.code?.description
+                cell.detailTextLabel!.text = "code"
+            default:
+                cell.textLabel!.text = ""
+                cell.detailTextLabel!.text = ""
+            }
+        case 5:
+            cell.textLabel!.text = self.observation.device?.reference?.description
+            cell.detailTextLabel!.text = "reference"
+        case 6:
+            switch(indexPath.row) {
+            case 0:
+                cell.textLabel!.text = observation.extension_fhir?.first?.valueCoding?.system?.description
+                cell.detailTextLabel!.text = "system"
+            case 1:
+                cell.textLabel!.text = observation.extension_fhir?.first?.valueCoding?.code?.description
+                cell.detailTextLabel!.text = "code"
+            case 2:
+                cell.textLabel!.text = observation.extension_fhir?.first?.valueCoding?.display?.description
+                cell.detailTextLabel!.text = "display"
+            case 3:
+                cell.textLabel!.text = observation.extension_fhir?.first?.url?.description
+                cell.detailTextLabel!.text = "url"
+            default:
+                ()
+            }
+        default:
+            print("")
+        }
+        
+        return cell
+    }
+    
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        if (observation.extension_fhir?.count) != nil {
+            return 7
+        }
+        
+        return 6
+    }
+    
+    override func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return 75
+    }
+    
+    override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        switch section {
+        case 0:
+            return "Code"
+        case 1:
+            return "Subject"
+        case 2:
+            return "Effective Period"
+        case 3:
+            return "Performer"
+        case 4:
+            return "Value Quantity"
+        case 5:
+            return "Device"
+        case 6:
+            return "Meal Context"
+        default:
+            return ""
+        }
+    }
+
+    //MARK: table delegate methods
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        print("didSelectRowAtIndexPath")
+        tableView.deselectRow(at: indexPath, animated: true)
+    }
+    
+    func refreshTable() {
```

Comment:
a great feature of swift is how protocols and extensions can provide nice defaults. e.g. you could add this `refresh()` method to any UITableViewController simply by adhereing to `Refreshable`. e.g.

```
protocol Refreshable {
    func refresh()
}

extension Refreshable where Self: UITableViewController {
    func refresh() {
        DispatchQueue.main.async {
            self.tableView.reloadData()
        }
    }
}

class FooTableViewController: UITableViewController, Refreshable {
    
    override func viewDidLoad() {
        refresh()
    }
    
}
```

And that saves you from having to write refreshTable in every table vc.